russstation/code/modules/vehicles/scooter.dm

```diff
@@ -0,0 +1,58 @@
+/obj/vehicle/ridden/wheelchair/russ/sportschair/Initialize()
+	. = ..()
+	sparks = new
+	sparks.set_up(1, 0, src)
+	sparks.attach(src)
+
+/obj/vehicle/ridden/wheelchair/russ/sportschair/Destroy()
+	if(sparks)
+		QDEL_NULL(sparks)
+	return ..()
+
+/obj/vehicle/ridden/wheelchair/russ/sportschair/generate_actions()
+	. = ..()
+		initialize_controller_action_type(/datum/action/vehicle/ridden/wheelchair/russ/sportschair/ollie, VEHICLE_CONTROL_DRIVE)
+
+/obj/vehicle/ridden/wheelchair/russ/sportschair/post_buckle_mob(mob/living/M)//allows skateboards to be non-dense but still allows 2 skateboarders to collide with each other
+	density = TRUE
+	return ..()
+
+/obj/vehicle/ridden/wheelchair/russ/sportschair/post_unbuckle_mob(mob/living/M)
+	if(!has_buckled_mobs())
+		density = FALSE
+	return ..()
+
+/obj/vehicle/ridden/wheelchair/russ/sportschair/Bump(atom/A)
+	. = ..()
+	if(!A.density || !has_buckled_mobs())
+		return
+
+///Moves the vehicle forward and if it lands on a table, repeats
+/obj/vehicle/ridden/wheelchair/russ/sportschair/proc/grind()
+	step(src, dir)
+	if(!has_buckled_mobs() || !(locate(/obj/structure/table) in loc.contents) && !(locate(/obj/structure/fluff/tram_rail) in loc.contents))
+		obj_flags = CAN_BE_HIT
+		grinding = FALSE
+		icon_state = "[initial(icon_state)]"
+		return
+
+	var/mob/living/L = buckled_mobs[1]
```

Comment:
Is there a reason this uses `buckled_mobs` instead of `owner` to get this, like as is done in the ollie code. Also is DM 1-indexed, or wouldn't this refer to the second buckled mob? In which case would it still grab the person using the wheelchair?